wp-includes/class-wp-customize-manager.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Customizer API: WP_Customize_Manager class
+ *
+ * Add control for maskable icon setting.
+ *
+ * @package PWA
+ * @subpackage Customize
+ * @since 0.7
+ */
+
+/**
+ * Register maskable icon setting.
+ *
+ * @param WP_Customize_Manager $wp_customize Customizer manager object.
+ *
+ * @return void
+ * @since 0.7
+ */
+function pwa_customize_register_maskable_icon_setting( WP_Customize_Manager $wp_customize ) {
+	$wp_customize->add_setting(
+		'pwa_maskable_icon',
+		array(
+			'capability' => 'manage_options',
+			'type'       => 'option',
```

Comment:
@westonruter `site_icon` is getting store in `*_options` table and maskable property is associated with `site_icon` itself. IMO, it makes sense to store `maskable` in `options` as there will be only one `site_icon` in customizer.

Reply:
But if someone picks a different image for the site icon, then it may not be maskable. So that's why I was thinking it could be useful to associate the “maskability“ with the image `attachment` post specifically. But this would add quite a bit of complexity and it isn't necessary as the user would clearly see that the icon is maskable or not upon selecting a new image.

Reply:
@westonruter Before your code changes, when user was picking new image for site icon, I was resetting the maskable toggle value.

Here, we can consider that every newly uploaded image may not be maskable and if user want he can explicitly mark the maskable input. 

Only drawback of this approach is next time user selects previously uploaded icon, check box will not be marked/unmarked based on last input for that icon.

IMO, considering the tradeoff of simplicity vs experience, for now we can proceed with `options` and may change it in future if such requests are raised.